lib/src/git_backend.rs

```diff
@@ -679,10 +681,9 @@ impl Backend for GitBackend {
 
     fn write_commit(&self, mut contents: Commit) -> BackendResult<(CommitId, Commit)> {
         let locked_repo = self.repo.lock().unwrap();
-        let git_tree_id = if let Some(tree_id) = contents.root_tree.as_resolved() {
-            validate_git_object_id(tree_id)?
-        } else {
-            write_tree_conflict(locked_repo.deref(), &contents.root_tree)?
+        let git_tree_id = match &contents.root_tree {
+            MergedTreeId::Legacy(tree_id) => validate_git_object_id(tree_id)?,
+            MergedTreeId::Merge(tree_ids) => write_tree_conflict(locked_repo.deref(), tree_ids)?,
```

Comment:
Should we build conflict tree even if `tree_ids` represents a non-legacy non-conflicting tree?

Reply:
No, we even have a test for that. I just also broke the test in this commit 🤦 Thanks for catching it.